compiler/modules/nback/function.bal

```diff
@@ -66,9 +66,16 @@ function buildCallIndirect(llvm:Builder builder, Scaffold scaffold, bir:CallIndi
     bir:Register funcOperand = insn.operands[0];
     var { funcValuePtr, funcDescPtr, funcPtr, uniformFuncPtr } = check buildIndirectFunctionValue(builder, scaffold, funcOperand);
     var [nArgs, uniformArgArray] = check buildUniformArgArray(builder, scaffold, insn);
-    t:FunctionAtomicType atomic = <t:FunctionAtomicType>t:functionAtomicType(scaffold.typeContext(),
-                                                                             funcOperand.semType);
-    // TODO: handle the case where this is not atomic (directly buildUniformCall)
+    t:SemType funcTy = funcOperand.semType;
+    t:FunctionAtomicType? atomic = t:functionAtomicType(scaffold.typeContext(), funcTy);
+    if atomic == () {
+        t:SemType[] argTypes = from var arg in args select arg.semType;
+        t:Context tc = scaffold.typeContext();
+        t:SemType returnType = <t:SemType>t:functionReturnType(scaffold.typeContext(), funcTy,
+                                                               t:tupleTypeWrappedRo(tc.env, ...argTypes));
+        return buildCallInexact(builder, scaffold, result, funcDescPtr, funcPtr, uniformFuncPtr,
+                                uniformArgArray, nArgs, returnType);
+    }
     t:FunctionSignature signature = t:functionSignature(scaffold.typeContext(), atomic);
```

Comment:
Have a comment here, saying that this case (non-constant, but atomic type) is the complicated one, because we decide at run-time whether we can do the shortcut or not.

Reply:
Fixed